algebird-core/src/main/scala/com/twitter/algebird/SketchMap.scala

```diff
@@ -50,7 +51,10 @@ class SketchMapMonoid[K, V](val params: SketchMapParams[K])(implicit
     SketchMap(AdaptiveMatrix.fill(params.depth, params.width)(monoid.zero), Nil, monoid.zero)
 
   override def plus(left: SketchMap[K, V], right: SketchMap[K, V]): SketchMap[K, V] = {
-    val newValuesTable = Monoid.plus(left.valuesTable, right.valuesTable)
+    val newValuesTable = right.valuesTable match {
+      case DenseMatrix(_, _, _) => Monoid.plus(right.valuesTable, left.valuesTable)
+      case _                    => Monoid.plus(left.valuesTable, right.valuesTable)
+    }
```

Comment:
```suggestion
    val newValuesTable = right.valuesTable match {
      case _: DenseMatrix[V] => Monoid.plus(right.valuesTable, left.valuesTable)
      case _                    => Monoid.plus(left.valuesTable, right.valuesTable)
    }
```

Also, perhaps it would make more sense to address this here https://github.com/twitter/algebird/blob/65a95ce440b9c30101db5528cf1a90ec02361142/algebird-core/src/main/scala/com/twitter/algebird/matrix/AdaptiveMatrix.scala#L56

Reply:
> Also, perhaps it would make more sense to address this here

yeah, pretty much agreed. 
I was afraid to touch `AdaptiveMatrix` due to unfamiliarity of it, but will try to see if I can fix that directly.